docs/src/configuration/app/source-operations.md

```diff
@@ -51,9 +51,39 @@ Note that these operations run in an isolated container which is not part of the
 
 Also, if multiple applications in a single project both result in a new commit, that will appear as two distinct commits in the Git history but only a single new build/deploy cycle will occur. If multiple applications define source operations with the same name, they will all be executed sequentially on each application.
 
+## Source Operations usage examples
+
+### Update a site from an upstream repository or template
+
+This example will synchronize your branch with the content from an upstream Git repository.
+
+You need to add a project-level variable named `env:UPSTREAM_REMOTE` with the Git URI of the upstream repository. That will make that repository available as a Unix environment variable in all environments, including in the Source Operations environment.
+
+Then, define a Source Operation to that upstream repository:
+
+```
+source:
+    operations:
+        upstream-update:
+            command: |
+                set -e
+                git remote add upstream $UPSTREAM_REMOTE
+                git fetch --all
+                git merge upstream/master
```

Comment:
```suggestion
                git merge upstream/main
```

Reply:
Today, all of our templates still all have their default branch called master. And that's the case with most customer upstream repos as well.

Reply:
I made it main to match the explanation below, the example is okay as long as it's consistent.